profile/prune_test.go

```diff
@@ -228,3 +230,50 @@ Locations
      5: 0x0 Foo::operator()(::Bar) fun.c:1 s=0
 Mappings
 `
+
+var locs3 = []*Location{
```

Comment:
Instead of keeping extending the test cases here, I'd add a table-based test for just simplifyFunc. It would be easier to add more test cases as simplifyFunc gets more complex.